dcc-portal-ui/app/scripts/pathwayviewer/js/directive.js

```diff
@@ -342,7 +334,16 @@
             return;
           }else if(!rendered){
             $('.pathwaysvg').remove();
-            controller.render(xml,zoomedOn);
+            
+            controllerSettings.model = new PathwayModel(xml);
+            controller = new ReactomePathway(controllerSettings);
+            controller.render(zoomedOn);
+
+            var legendSVGElement = controller.getLegendSVGElement(270, 671);
+            d3.select('.pathway-legend-content').append(function(){
+              return legendSVGElement;
```

Comment:
Would this still work if we directly passed `legendSVGElement` to `append()`? I don't see why it needs to be wrapped in a function